test/walletsReducer.js

```diff
@@ -0,0 +1,223 @@
+const { assert } = require('chai');
+const rewire = require('rewire');
+
+const {
+  WALLETS_NAMESPACE,
+  MULTISIG_WALLETS_NAMESPACE,
+  ACCOUNTS_NAMESPACE,
+  ACCOUNT_INFO_NAMESPACE,
+  WALLET_INFO_NAMESPACE,
+  BALANCE_NAMESPACE,
+  HISTORY_NAMESPACE,
+  SELECTED_ACCOUNT_NAMESPACE,
+  SELECTED_WALLET_NAMESPACE,
+  PLUGIN_NAMESPACE,
+  MULTISIG_WALLET_INFO_NAMESPACE,
+  MULTISIG_PROPOSAL_NAMESPACE,
+  SELECTED_PROPOSAL_NAMESPACE,
+} = require('../lib/constants');
+
+const { reduceWallets, [PLUGIN_NAMESPACE]: reduceApp } = require('../lib/reducers');
+
+const walletResponses = require('./data/walletResponses.json');
+
+const {
+  getWalletResponse,
+  getMultisigResponse,
+  getWalletInfoResponse,
+  createWalletResponse,
+  createMSWalletResponse,
+  getAccountsResponse,
+  getAccountInfoResponse,
+  getAccountBalanceResponse,
+  getAccountHistoryResponse,
+  createAccountResponse,
+  getProposalMTXResponse,
+} = walletResponses;
+
+const { getClientMock } = require('./mockClient.js');
+
+// monkey patch getClient function
+const walletActions = rewire('../lib/actions/wallet');
```

Comment:
wow, nice. this looks like a super useful testing utility.

Reply:
The alternative option would be to use dependency injection such that each method accepts a client, then we wouldn't need to use this library

Reply:
Another one actually I think could be to just stub the methods themselves using a library like `sinon` and confirm that they get called without actually calling the methods. This would mean the tests won't check any more if the final result is correct, though you could stub the results using the same fake client you have now. `rewire` seems super convenient though.

Reply:
I think that checking to make sure that the correct place in the store has been updated is important